kedro-datasets/kedro_datasets/databricks/managed_table_dataset.py

```diff
@@ -0,0 +1,463 @@
+"""``ManagedTableDataSet`` implementation to access managed delta tables
+in Databricks.
+"""
+import logging
+import re
+from dataclasses import dataclass
+from functools import partial
+from operator import attrgetter
+from typing import Any, Dict, List, Union
+
+import pandas as pd
+from cachetools import Cache, cachedmethod
+from cachetools.keys import hashkey
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    DataSetError,
+    Version,
+    VersionNotFoundError,
+)
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+from pyspark.sql.utils import AnalysisException, ParseException
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ManagedTable:  # pylint: disable=R0902
+    """Stores the definition of a managed table"""
+
+    # regex for tables, catalogs and schemas
+    _NAMING_REGEX = r"\b[0-9a-zA-Z_-]{1,32}\b"
```

Comment:
We incorporated dash support in naming regex with the following: 

`    _NAMING_REGEX = r"\b[0-9a-zA-Z_\-]{1,32}\b"
`